python/dolfinx/wrappers/fem.cpp

```diff
@@ -474,7 +474,7 @@ void fem(py::module& m)
       "locate_dofs_geometrical",
       [](const std::vector<
              std::reference_wrapper<const dolfinx::fem::FunctionSpace>>& V,
-         const std::function<Eigen::Array<bool, Eigen::Dynamic, 1>(
+         const std::function<std::vector<bool>(
```

Comment:
In all python interfaces (also below), we need to have `py::array` of bools, for performance reasons.

Reply:
`std::vector` allocates its own memory, and doesn't have a `reference` constructor.
I'm having trouble understanding how it could help.

Reply:
> `std::vector` allocates its own memory, and doesn't have a `reference` constructor.
> I'm having trouble understanding how it could help.

We might have to make a copy in the wrappers, which is probably ok. The point @michalhabera is making is that pybind11 is really slow in handling `list` <-> `std::vector`, so it needs to managed explicitly.